src/memphis.ts

```diff
@@ -11,765 +11,553 @@
-   * @param {Number} replicas - number of replicas for the messages of the data, default is 1.
-   * @param {Boolean} dedupEnabled - whether to allow dedup mecanism, dedup happens based on message ID, default is false.
-   * @param {Number} dedupWindowMs - time frame in which dedup track messages, default is 0.
-   */
-  async station({
-    name,
-    factoryName,
-    retentionType = retentionTypes.MAX_MESSAGE_AGE_SECONDS,
-    retentionValue = 604800,
-    storageType = storageTypes.FILE,
-    replicas = 1,
-    dedupEnabled = false,
-    dedupWindowMs = 0,
-  }: {
-    name: string;
-    factoryName: string;
-    retentionType?: string;
-    retentionValue?: number;
-    storageType?: string;
-    replicas?: number;
-    dedupEnabled?: boolean;
-    dedupWindowMs?: number;
-  }): Promise<Station> {
-    try {
-      if (!this.isConnectionActive) throw new Error("Connection is dead");
-      const response = await httpRequest({
-        method: "POST",
-        url: `http://${this.host}:${this.managementPort}/api/stations/createStation`,
-        headers: {
-          Authorization: "Bearer " + this.accessToken,
-        },
-        bodyParams: {
-          name: name,
-          factory_name: factoryName,
-          retention_type: retentionType,
-          retention_value: retentionValue,
-          storage_type: storageType,
-          replicas: replicas,
-          dedup_enabled: dedupEnabled,
-          dedup_window_in_ms: dedupWindowMs,
-        },
-      });
-
-      return new Station(this, response.name);
-    } catch (ex) {
-      if (typeof ex == "string") {
-        return new Station(this, name.toLowerCase());
-      }
-      throw ex;
+
+    private _normalizeHost(host: string): string {
+        if (host.startsWith('http://')) return host.split('http://')[1];
+        else if (host.startsWith('https://')) return host.split('https://')[1];
+        else return host;
+    }
+
+    /**
+     * Creates a factory.
+     * @param {String} name - factory name.
+     * @param {String} description - factory description (optional).
+     */
+    async factory({ name, description = '' }: { name: string; description?: string }): Promise<Factory> {
+        try {
+            if (!this.isConnectionActive) throw new Error('Connection is dead');
+
+            let createFactoryReq = {
+                username: this.username,
+                factory_name: name,
+                factory_description: description
+            };
+            let data = this.JSONC.encode(createFactoryReq);
+            let errMsg = await this.brokerManager.request('$memphis_factory_creations', data);
+            errMsg = errMsg.data.toString();
+            if (errMsg != '') {
+                throw new Error(errMsg);
+            }
+            return new Factory(this, name);
+        } catch (ex) {
+            if (typeof ex == 'string') {
```

Comment:
this logic should handle cases were this factory is already exist and if it does ignoring the exception, so now we switched from http to talk over an internal subject, so please simulate a case of creating a factory which is already exist, catch this error and for that specific error - ignore, al other errors should be handled as you did